### ISSUE TITLE
cmd/go: building cgo, "unknown relocation type 42; compiled without -fpic?"

### ISSUE THREAD

**Issue by KALRONG** [golang/go#13896]
hi,

While trying to build go from sources on Debian 8 Stretch I keep getting the following error and the build fails:

/usr/src/go/pkg/linux_amd64/runtime/cgo.a(_all.o): unknown relocation type 42; compiled without -fpic?

So far I couldn't find any workaround for this issue.

**Comment by davecheney** [comment]
I believe this is a binutils bug, it was reported recently by a fedora user and has been escalated to binutils.

**Comment by KALRONG** [comment]
Thanks a lot, do you know if there is any workaround in the meantime? maybe another version of binutils?

**Comment by mwhudson** [comment]
Which version are you building? If it's not the thing Dave's talking about, 1.5.2 and tip should build OK.

**Comment by KALRONG** [comment]
I tried 1.5.2 and 1.5.3 and had the same issue in both so it seems to be that binutils bug, will wait for debian to get an update

**Comment by bradfitz** [comment]
@ianlancetaylor, ring any bells?

Also, are we missing builder coverage that we missed this? @KALRONG, what version of binutils do you have?

**Comment by ianlancetaylor** [comment]
This is a bug in the version of the GNU binutils used on some very up to date Debian systems.  Relocation type 42 is being generated by the C assembler but is not being recognized by the C linker.  There is nothing that Go can do to fix this.

**Comment by bradfitz** [comment]
http://linux.debian.bugs.dist.narkive.com/koUKCmST/bug-807136-golang-ftbfs-due-to-new-binutils-relocations suggests this was fixed in Go 1.5.2 via git rev 914db9f060b1f? But @KALRONG reports otherwise?

Anyway, I'm happy if @ianlancetaylor is happy.

**Comment by ianlancetaylor** [comment]
@bradfitz You're right, I did misread the bug.  This was a bug in Go, not in the binutils.  But I'm reasonably certain that that bug is fixed.  @KALRONG are you really sure you still see it?

**Comment by bradfitz** [comment]
This has been fixed.

I installed a new Debian scratch system in a VM and reproduced the failure with Go 1.5.1, then updated to Go 1.5.2 and it worked fine.

@KALRONG, just use Go 1.5.2+ (1.5.3 is recommended). You must have had a confused path/environment to Go 1.5.n.

<img width="829" alt="screen shot 2016-01-21 at 6 44 25 pm" src="https://cloud.githubusercontent.com/assets/2621/12501158/086fd280-c06f-11e5-92b7-51a7615f6c46.png">

**Comment by KALRONG** [comment]
Hi, 

Sorry for the delay on replying as I was on vacation and didn't get to check this.

I have done and upgrade on my debian (strecth) and tried compiling again and this time no errors appeared and everything is working as it should with 1.5.3. I didn't change anything else, just updated the debian and then compile from sources. 

The only thing I saw is that one of the packages updated was the debian package for go, my guess is that this package in debian (being testing) still had this bug and while I tried to compile from source (needed to add more options than the ones in the debian package) this error showed. This seems to have been fixed on the debian side and now Im able to compile go from source without issues.

Thanks everybody for looking in to this, and sorry for the trouble.
